src/java/org/pantsbuild/tools/ivy/DependencyUpdateChecker.java

```diff
@@ -0,0 +1,295 @@
+              log(sb.toString());
+              dependencyUpdateDetected = true;
+            }
+          }
+        }
+      }
+    }
+    if (!dependencyUpdateDetected) {
+      log(INDENTATION + "All dependencies are up to date");
+    }
+  }
+
+  private void displayMissingDependencyOnLatest(ResolveReport originalReport,
+      ResolveReport latestReport) {
+    List<ModuleRevisionId> listOfMissingDependencyOnLatest = new ArrayList<>();
+    for (Object ivyNode : originalReport.getDependencies()) {
+      IvyNode originalDependency = (IvyNode) ivyNode;
+      boolean dependencyFound = false;
+      for (Object ivyNode2 : latestReport.getDependencies()) {
+        IvyNode latest = (IvyNode) ivyNode2;
+        if (originalDependency.getModuleId().equals(latest.getModuleId())) {
+          dependencyFound = true;
+        }
+      }
+      if (!dependencyFound) {
+        listOfMissingDependencyOnLatest.add(originalDependency.getId());
+      }
+    }
+
+    if (listOfMissingDependencyOnLatest.size() > 0) {
+      log("List of missing dependencies on latest resolve:");
+      for (ModuleRevisionId moduleRevisionId : listOfMissingDependencyOnLatest) {
+        log(INDENTATION + moduleRevisionId.toString());
+      }
+    }
+  }
+
+  private void displayNewDependencyOnLatest(ResolveReport originalReport,
+      ResolveReport latestReport) {
+    List<ModuleRevisionId> listOfNewDependencyOnLatest = new ArrayList<>();
+    for (Object ivyNode : latestReport.getDependencies()) {
+      IvyNode latest = (IvyNode) ivyNode;
+
+      boolean dependencyFound = false;
+      for (Object ivyNode2 : originalReport.getDependencies()) {
+        IvyNode originalDependency = (IvyNode) ivyNode2;
+        if (originalDependency.getModuleId().equals(latest.getModuleId())) {
+          dependencyFound = true;
+        }
+      }
+      if (!dependencyFound) {
+        listOfNewDependencyOnLatest.add(latest.getId());
+      }
+    }
+    if (listOfNewDependencyOnLatest.size() > 0) {
+      log("List of new dependencies on latest resolve:");
+      for (ModuleRevisionId moduleRevisionId : listOfNewDependencyOnLatest) {
+        log(INDENTATION + moduleRevisionId.toString());
+      }
+    }
+  }
+
+  public Ivy getIvyInstance() throws ParseException, IOException {
+    if (ivyInstance != null) {
+      return ivyInstance;
+    }
+
+    ivyInstance = Ivy.newInstance();
+
+    URLHandlerDispatcher dispatcher = new URLHandlerDispatcher();
+    URLHandler httpHandler = URLHandlerRegistry.getHttp();
+    dispatcher.setDownloader("http", httpHandler);
+    dispatcher.setDownloader("https", httpHandler);
+    URLHandlerRegistry.setDefault(dispatcher);
+
+    IvySettings settings = ivyInstance.getSettings();
+    settings.addAllVariables(System.getProperties());
+
+    ivyInstance.configure(settingsFile);
+
+    return ivyInstance;
+  }
+
+  public ResolveReport getResolvedReport() throws ParseException, IOException {
```

Comment:
Memoizing like this is a bit odd. Would seem more straightforward to just calculate the report once and hold it in a local variable rather than calling this repeatedly.